openedx/core/djangoapps/theming/finders.py

```diff
@@ -17,63 +17,80 @@
 .. _Django-Pipeline: http://django-pipeline.readthedocs.org/
 .. _Django-Require: https://github.com/etianen/django-require
 """
-from path import Path
-from django.conf import settings
-from django.core.exceptions import ImproperlyConfigured
+import os
+from collections import OrderedDict
+
 from django.contrib.staticfiles import utils
 from django.contrib.staticfiles.finders import BaseFinder
-from openedx.core.djangoapps.theming.storage import CachedComprehensiveThemingStorage
+from django.utils import six
+
+from openedx.core.djangoapps.theming.helpers import get_themes
+from openedx.core.djangoapps.theming.storage import ThemeStorage
 
 
-class ComprehensiveThemeFinder(BaseFinder):
+class ThemeFilesFinder(BaseFinder):
     """
-    A static files finder that searches the active comprehensive theme
-    for static files. If the ``COMPREHENSIVE_THEME_DIR`` setting is unset,
-    or the ``COMPREHENSIVE_THEME_DIR`` does not exist on the file system,
-    this finder will never find any files.
+    A static files finder that looks in the directory of each theme as
+    specified in the source_dir attribute.
     """
+    storage_class = ThemeStorage
+    source_dir = 'static'
```

Comment:
I think that would cause confusion given that we already have `STATIC_ROOT` and `STATICFILES_DIRS` and the new setting will only be applicable for `ThemeFilesFinder` (this might cause unexpected behavior if someone updates the setting).